src/gridcoin/voting/registry.cpp

```diff
@@ -311,6 +317,266 @@ int64_t PollReference::Expiration() const
     return m_timestamp + (m_duration_days * 86400);
 }
 
+CBlockIndex* PollReference::GetStartingBlockIndexPtr() const
+{
+    uint256 block_hash;
+    CTransaction tx;
+
+    GetTransaction(*m_ptxid, tx, block_hash);
+
+    if (!block_hash.IsNull()) {
+        CBlockIndex* pindex = mapBlockIndex[block_hash];
+
+        return pindex;
+    }
+
+    return nullptr;
+}
+
+CBlockIndex* PollReference::GetEndingBlockIndexPtr() const
+{
+    // Has poll ended?
+    if (Expiration() < GetAdjustedTime()) {
+        GRC::BlockFinder blockfinder;
+
+        // Find the last block that contains valid votes for the poll.
+        CBlockIndex* pindex = blockfinder.FindByMinTime(Expiration());
+
+        return pindex;
+    }
+
+    return nullptr;
+}
+
+std::optional<int> PollReference::GetStartingHeight() const
+{
+    const CBlockIndex* pindex = GetStartingBlockIndexPtr();
+
+    if (pindex != nullptr) {
+        return pindex->nHeight;
+    }
+
+    return std::nullopt;
+}
+
+std::optional<int> PollReference::GetEndingHeight() const
+{
+    CBlockIndex* pindex = GetEndingBlockIndexPtr();
+
+    if (pindex != nullptr) {
+        return pindex->nHeight;
+    }
+
+    return std::nullopt;
+}
+
+std::optional<CAmount> PollReference::GetActiveVoteWeight() const
+{
+    // Instrument this so we can log real time performance.
+    g_timer.InitTimer(__func__, LogInstance().WillLogCategory(BCLog::LogFlags::VOTE));
+
+    // Get the start and end of the poll.
+    CBlockIndex* pindex_start = GetStartingBlockIndexPtr();
+    const CBlockIndex* pindex_end = GetEndingBlockIndexPtr();
```

Comment:
nit?
```suggestion
    const CBlockIndex* const pindex_start = GetStartingBlockIndexPtr();
    CBlockIndex* const pindex_end = GetEndingBlockIndexPtr();
```

Reply:
It has to be this way...
    const CBlockIndex* const pindex_start = GetStartingBlockIndexPtr();
    const CBlockIndex* pindex_end = GetEndingBlockIndexPtr();
because pindex_end is conditionally reassigned to pindexBest if originally nullptr.

Reply:
It really isn't that important here anyway, since these are all local variables inside the function.

Reply:
Oops, I mixed up the place for const for `pindex_end`.

The `pindex_start` could be `const CBlockIndex* const` if you also add const to the loops below but yeah it's not really important.